### PR TITLE
Rename example test files, and add requirement for new recipes.

### PR DESCRIPTION
**Pull request by sawsa307**
* Rename test_example and its files to avoid cleanup.sh template being picked up by cleanup_all.sh.
* Add warning on break changes will be reverted.
* Add requirement for new recipe PR, and example for the required output.

### CODE REVIEW

File: test/README.md [hunk 1]
```diff
@@ -18,6 +18,8 @@ limitations under the License.
 
 The tests here are intended to test the recipes in this repo to make sure it workable and up-to-date.
 
+PLEASE NOTE: All recipe tests are run periodically to ensure they are all passed. If a new recipe or changes to an existing recipe causes test failures, it will be REVERTED.
```
Comment (bowei) [inline test/README.md:21]: It may be reverted.

File: test/README.md [hunk 2]
```diff
@@ -75,7 +77,7 @@ To cleanup all tests separately, use the following command from test/:
 
 ## Adding a new recipe test
 
-For a new recipe, in addition to its yaml file and REAME.md, it should also include a set of test files to make sure the recipe is functional and up-to-date.
+For a new recipe, in addition to its yaml file and REAME.md, it should also include a set of test files to make sure the recipe is functional and up-to-date. In the description section of the pull request, you should also provide the result of `make test` tp show your test is passing and is not breaking any other existing tests. See example in [output-example.txt](./test_example/output_example.txt).
```
Comment (bowei) [inline test/README.md:80]: tp show => to show

any other existing => other tests

**Review by bowei — APPROVED** [review]

File: test/README.md (hunk 1 above)
Comment (sawsa307) [reply test/README.md]: Updated.

File: test/README.md (hunk 2 above)
Comment (sawsa307) [reply test/README.md:80]: Updated.

**Comment by sawsa307** [comment]
Hi Bowei I've addressed the typos, could you take another look?

**Review by bowei — APPROVED** [review]

**Pull request merged**